python/src/lib.rs

```diff
@@ -339,50 +366,73 @@ impl RawDeltaTable {
         min_commit_interval = None,
         writer_properties=None,
         custom_metadata=None,
+        commit_writes=None
     ))]
+    #[allow(clippy::too_many_arguments)]
     pub fn compact_optimize(
         &mut self,
+        py: Python,
         partition_filters: Option<Vec<(&str, &str, PartitionFilterValue)>>,
         target_size: Option<i64>,
         max_concurrent_tasks: Option<usize>,
         min_commit_interval: Option<u64>,
         writer_properties: Option<HashMap<String, Option<String>>>,
         custom_metadata: Option<HashMap<String, String>>,
+        commit_writes: Option<bool>,
     ) -> PyResult<String> {
-        let mut cmd = OptimizeBuilder::new(
-            self._table.log_store(),
-            self._table.snapshot().map_err(PythonError::from)?.clone(),
-        )
-        .with_max_concurrent_tasks(max_concurrent_tasks.unwrap_or_else(num_cpus::get));
-        if let Some(size) = target_size {
-            cmd = cmd.with_target_size(size);
-        }
-        if let Some(commit_interval) = min_commit_interval {
-            cmd = cmd.with_min_commit_interval(time::Duration::from_secs(commit_interval));
-        }
+        py.allow_threads(|| {
+            let mut cmd = OptimizeBuilder::new(
+                self._table.log_store(),
+                self._table.snapshot().map_err(PythonError::from)?.clone(),
+            )
+            .with_max_concurrent_tasks(max_concurrent_tasks.unwrap_or_else(num_cpus::get));
+            if let Some(size) = target_size {
+                cmd = cmd.with_target_size(size);
+            }
 
-        if let Some(writer_props) = writer_properties {
-            cmd = cmd.with_writer_properties(
-                set_writer_properties(writer_props).map_err(PythonError::from)?,
-            );
-        }
+            if let Some(commit_interval) = min_commit_interval {
+                cmd = cmd.with_min_commit_interval(time::Duration::from_secs(commit_interval));
 
-        if let Some(metadata) = custom_metadata {
-            let json_metadata: Map<String, Value> =
-                metadata.into_iter().map(|(k, v)| (k, v.into())).collect();
-            cmd = cmd.with_metadata(json_metadata);
-        };
+                if let Some(commit_optimize_writes) = commit_writes {
+                    if !commit_optimize_writes && commit_interval > 0 {
+                        warn!("Commit interval is > 0, but commit_writes is false, so delta-rs will not commit")
```

Comment:
this warning log seems spammy, if we expect this to happen in nominal cases, then we shouldn't log it as warn. in general, it's better to let python do the logging if possible so the logging level can be controlled on the python side.

Reply:
Decided it's enough to just document the behavior, if the user deliberately passes in `write_commit = false` I think it's reasonable to expect them to understand that the optimize will not commit.